Makefile.am

```diff
@@ -276,23 +276,8 @@ cc_proxy_extra_dist =			\
 
 CHECK_DEPS += check-proxy
 
-define CONTENT
-package main
-import "fmt"
-func main() {
-	fmt.Printf("Executing test program")
-}
-endef
-export CONTENT
-
 check-proxy:
-	$(AM_V_GEN)go_test_extra_args="" ; \
-	echo "$$CONTENT">./racetest.go ; \
-	if go build -race ./racetest.go; then \
-		go_test_extra_args+="-race" ; \
-	fi ; \
-	rm ./racetest* ; \
-	go test -v "$$go_test_extra_args" -timeout 2s $(srcdir)/proxy
+	$(AM_V_GEN)go test -v -race -timeout 2s $(srcdir)/proxy || go test -v -timeout 2s $(srcdir)/proxy
```

Comment:
That's a lot simpler and neater! I'd probably make it even simpler though by using variables to avoid hard-coding `-timeout 2s $(srcdir)/proxy` twice as that makes maintenance more difficult.